### PR TITLE
Run tests

### PR DESCRIPTION
**Pull request by freddydk**
Add a new workflow to be able to run selected tests on a given REF, like:

image

### CODE REVIEW

File: .github/workflows/RunTests.yaml [hunk 1]
```diff
@@ -0,0 +1,158 @@
+name: Run Tests
+
+on:
+  workflow_dispatch:
+    inputs:
+      ref:
+        description: ref on which to run the end-2-end tests (default is head_sha on main)
```
Comment (mazhelez) [inline .github/workflows/RunTests.yaml:7]: NAVContainerHelper's default branch is _master_.

Also, isn't the default the branch on which the workflow is dispatched?

File: .github/workflows/RunTests.yaml [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+              Write-Host "::Error::$($result.FailedCount) tests are failing"
+              $host.SetShouldExit(1)
+            }
+          }
+          catch {
+            Write-Host "::Error::Exception when running tests. The Error was $($_.Exception.Message)"
+            $host.SetShouldExit(1)
+          }
+
+  PS7:
```
Comment (mazhelez) [inline .github/workflows/RunTests.yaml:90]: I'd name this one 'Windows PS Core' and the next one 'Linux PS Core'

Comment (freddydk) [reply .github/workflows/RunTests.yaml:90]: It is a copy of CI/CD - will keep them the same and rename when making CI/CD call this one or like

File: .github/workflows/RunTests.yaml (hunk 1 above)
Comment (freddydk) [reply .github/workflows/RunTests.yaml]: fixed

File: .github/workflows/RunTests.yaml [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+        with:
+          ref: ${{ github.event.inputs.ref }}
+          lfs: true
+    
+      - name: Run Tests
+        run: |
+          try {
+            $errorActionPreference = "stop"
+            Set-StrictMode -version 2.0
+            $pesterContainer = New-PesterContainer -Path (Join-Path $ENV:GITHUB_WORKSPACE 'Tests\${{ matrix.test }}.ps1') -Data @{ "licenseFile" = '${{ secrets.licensefile }}'; "buildLicenseFile" = '${{ secrets.buildLicenseFile }}'; "insiderSasToken" = '${{ secrets.insiderSasToken }}' }
```
[the remark this answers is not in the document]
Comment (freddydk) [reply .github/workflows/RunTests.yaml:78]: Will check

[the remark this answers is not in the document]
Comment (freddydk) [reply .github/workflows/RunTests.yaml]: Have removed insider sas token from all tests - will re-run tests now

**Comment by freddydk** [comment]
Tests are passing without the insider sas token in the scripts (and secret removed)

**Review by mazhelez — APPROVED** [review]

**Pull request merged**